src/main/java/io/github/intellij/dlanguage/psi/DlangFile.java

```diff
@@ -122,37 +116,6 @@ public boolean processDeclarations(@NotNull final PsiScopeProcessor processor, @
         return toContinue;
     }
 
-    public DlangFunctionDeclaration getMainFunction() {
-        final DlangFunctionDeclaration[] res = new DlangFunctionDeclaration[1];
-        PsiScopeProcessor mainFunctionProcessor = new PsiScopeProcessor() {
-
-            @Override
-            public boolean execute(@NotNull PsiElement element, @NotNull ResolveState state) {
-                if (element instanceof DlangFunctionDeclaration) {
-                    if (((DNamedElement) element).getName().equals("main")) {
-                        res[0] = (DlangFunctionDeclaration) element;
-                        return false;
-                    }
-                }
-                return true;
-            }
-
-            @Nullable
-            @Override
-            public <T> T getHint(@NotNull Key<T> hintKey) {
-                return null;
-            }
-
-            @Override
-            public void handleEvent(@NotNull Event event, @Nullable Object associated) {
-
-            }
-        };
-        this.processDeclarations(mainFunctionProcessor, ResolveState.initial(), null, this);
-        return res[0];
-
-    }
-
```

Comment:
why is this `getMainFunction()` method removed? Are you sure it's not used? There are times that Intellij thinks a method is unused even when it is.

Reply:
Will check with simple text search. Would that suffice, or does idea use some kind of reflection on these files?

Reply:
Just checking with search should be fine

Reply:
ok, checked, and seems that all removed ones are used nowhere.